buildSrc/src/main/java/org/elasticsearch/gradle/VersionCollection.java

```diff
@@ -228,7 +229,7 @@ private String getBranchFor(Version version) {
         }
     }
 
-    public List<Version> getUnreleased() {
+    protected List<Version> getUnreleased() {
```

Comment:
We reduced visibility on this, but we still use this method if a couple of places outside this class and its tests. We now have this weirdness where `VersionCollections.unreleased` and `VersionCollections.getUnreleased` are different sets. Maybe we should rename the former to make it clear it includes on index compatible versions? We might also consider refactoring external usages.